onnxruntime/core/framework/session_state.cc

```diff
@@ -459,7 +463,9 @@ const NodeIndexInfo& SessionState::GetNodeIndexInfo() const {
 }
 
 void SessionState::UpdateToBeExecutedNodes(const std::vector<int>& fetch_mlvalue_idxs) {
-  if (to_be_executed_nodes_.find(fetch_mlvalue_idxs) != to_be_executed_nodes_.end())
+  std::vector<int> sorted_idxs = fetch_mlvalue_idxs;
+  std::sort(sorted_idxs.begin(), sorted_idxs.end());
+  if (to_be_executed_nodes_.find(sorted_idxs) != to_be_executed_nodes_.end())
```

Comment:
>to_be_executed_nodes_ [](start = 6, length = 21)

We are using a vector as the key for the map... 
How's the hash function composed?

Reply:
it's a std::map, vector has comparison operators defined